tests/test_ir.py

```diff
@@ -162,6 +162,39 @@ def test_region_clone_into_circular_blocks():
     assert region.is_structurally_equivalent(region2)
 
 
+def test_non_empty_block_with_parent_region_requires_terminator():
+    """
+    Tests that an empty block belonging to a single-block region with parent
+    operation requires terminator operation.
```

Comment:
Can you add a similar test but without the last operation having a successor?
I think you took the assumption in your code that the terminator must have a successor, however that's not the case.

Reply:
And maybe a test that has a terminator at the end, and that work (I would just cut paste the one in `test_builtin_traits`).

Reply:
Regarding the first comment, done.
I don't understand your second comment.

Reply:
Right now, you are checking that the code fails if there is no terminator at the end.
I think you should also check the positive case, which is a simple block with a terminator, where the verifier doesn't fail.
This is just to be sure we test both cases here.
You already have the test in `test_builtin_traits`, but I think you should copy it there.
Does that makes sense?

Reply:
hmm, yeah, let me see

Reply:
I've added the test case named `test_non_empty_block_with_single_block_parent_region_can_have_terminator`.
I think this covers this.